bakery/src/tests/test_cli.js

```diff
@@ -16,6 +27,75 @@ const completion = subprocess => {
   })
 }
 
+const formatSubprocessOutput = (result) => {
+  return dedent`
+  ###### stdout ######
+  ${result.stdout}
+  ###### stderr ######
+  ${result.stderr}
+  `
+}
+
+test('build pipelines', async t => {
+  const envs = (await fs.readdir('env')).map(file => path.basename(file, '.json'))
+  const pipelines = (await fs.readdir('src/pipelines')).map(file => path.basename(file, '.js'))
+
+  const processes = []
+  for (const pipeline of pipelines) {
+    for (const env of envs) {
+      processes.push(
+        completion(spawn('./build', [
+          'pipeline',
+          pipeline,
+          env
+        ]))
+      )
+    }
+  }
+
+  await Promise.all(processes)
+  t.pass()
+})
+
+test('pin pipeline tasks to versions', async t => {
+  const sourceObjs = (obj) => {
+    const sources = []
+    if (typeof obj !== 'object') { return sources }
+    if (obj instanceof Array) {
+      for (const subobj of obj) {
+        sources.push(...sourceObjs(subobj))
+      }
+    }
+    if (obj.type != null && obj.type === 'docker-image') {
+      sources.push(obj.source)
+    } else {
+      for (const key of Object.keys(obj)) {
+        sources.push(...sourceObjs(obj[key]))
+      }
+    }
+    return sources
+  }
+
+  const customTag = 'my-custom-tag'
+  let pipelineOut = ''
+  const buildPipeline = spawn('./build', [
+    'pipeline',
+    'pdf',
+    'prod',
+    `--tag=${customTag}`
+  ])
+  buildPipeline.stdout.on('data', (data) => {
+    pipelineOut += data.toString()
+  })
+  const buildPipelineResult = await completion(buildPipeline)
+
+  const obj = yaml.safeLoad(pipelineOut)
+  const sources = sourceObjs(obj)
+  for (const source of sources) {
+    t.is(source.tag, customTag, formatSubprocessOutput(buildPipelineResult))
+  }
+})
+
 test('stable flow in content distribution pipeline', async t => {
```

Comment:
Minor: Update the test label text perhaps (since it now runs both dist and PDF tests)?